test/test_nestedtensor.py

```diff
@@ -549,6 +552,109 @@ def test_nested_tensor_indexing(self, device, dtype):
         expected_grad = torch.nested_tensor([grad_x0, torch.zeros((3, 4), device=device, dtype=dtype)])
         self.assertEqual(nt.grad, expected_grad)
 
+    @dtypes(*floating_types_and_half())
+    def test_nested_tensor_slicing(self, device, dtype):
+        # Create NT with one regular dimension, one jagged dimension
+        x = torch.nested_tensor([
+            torch.randn(2, 8, device=device, dtype=dtype),
+            torch.randn(3, 8, device=device, dtype=dtype),
+            torch.randn(4, 8, device=device, dtype=dtype)
+        ])
+
+        # Success case: slicing NT components (dim=0)
+        output = x[1:]
+        for i in range(1, x.size(0)):
+            self.assertEqual(x[i], output[i - 1])
+
+        output = x[:2]
+        for i in range(0, 2):
+            self.assertEqual(x[i], output[i])
+
+        # Success case: slicing regular dimension
+        NUM_CHUNKS = 4
+        CHUNK_SIZE = x.size(-1) // NUM_CHUNKS
+        output = x.chunk(NUM_CHUNKS, dim=2)
+        self.assertEqual(len(output), NUM_CHUNKS)
+        for chunk_idx, nt_chunk in enumerate(output):
+            self.assertTrue(nt_chunk.is_nested)
+
+            # Ensure same number of underlying tensor components
+            self.assertEqual(nt_chunk.dim(), x.dim())
+            self.assertEqual(nt_chunk.dim(), x.dim())
+
+            for i, component in enumerate(nt_chunk):
+                # Compare to manual slicing
+                start = chunk_idx * CHUNK_SIZE
+                end = start + CHUNK_SIZE
+                expected = x[i][:, start:end]
+                self.assertEqual(component, expected)
+
+        # Success case: chunking empty NT results in N chunks of original shape
+        empty = torch.nested_tensor([
+            torch.randn(0, 2, device=device, dtype=dtype),
+            torch.randn(0, 3, device=device, dtype=dtype),
+        ])
+        output = empty.chunk(NUM_CHUNKS, dim=1)
+        self.assertEqual(len(output), NUM_CHUNKS)
+        for nt_chunk in output:
+            self.assertEqual(nt_chunk.numel(), 0)
+            for i, component in enumerate(nt_chunk):
+                self.assertEqual(empty[i].shape, component.shape)
+
+        # Error case: slicing jagged dimension
+        with self.assertRaisesRegex(RuntimeError, 'dimension 1 is irregular'):
+            x.narrow(1, 1, 2)
```

Comment:
Just for future reference, this doesn't *have* to fail. We could easily narrow along an irregular dimension as long as there's enough data, it'll just become regular as a result.